surfer/viz.py

```diff
@@ -932,6 +932,7 @@ def add_overlay(self, source, min=2, max="robust_max", sign="abs",
             If None, it is assumed to belong to the hemipshere being
             shown. If two hemispheres are being shown, an error will
             be thrown.
+        kwargs: other mayavi surface arguments.
```

Comment:
This should follow the NumPy docstring format, like:
```
kwargs : dict
    Other Mayavi surface arguments.
```

Reply:
But it is not necessarily to be a dict here. How about changing it to "All additional keyword arguments are passed to the
        `mlab.pipeline.surface` call." and specifying the name `kwargs` to `mlab_kws`?
Or I could just remove the `**` symbols to make it a real dict?

Reply:
Let's do what nibabel does (except we don't need to escape the start characters)

https://github.com/nipy/nibabel/pull/827/files#diff-a1d0ba69368622536897ef9fe4aee05eR766

Reply:
This link is better (sorry on mobile)

https://github.com/nipy/nibabel/blob/ddab5468b12535b044b11837ee2776bb989bc341/nibabel/arraywriters.py#L266

Reply:
There are a lot of detailed arguments for each mayavi function, so I think it is hard to list all the possible arguments. Could we just list some useful examples?

Reply:
And one more thing, I think we should use the same name for the same argument in different functions. For example, the `alpha` in `add_data()` in mayavi is called `opacity`, so I think we should also specify this name in other functions and avoid it to appear in `mlab_kws`.

Reply:
I don't think we need to list all mayavi keywords, but it would be great if you could enable intersphinx so that the docs produce a link to the relevant mayavi api doc.

Nor do I think that we should have `mlab_kws`; the convention is to collect extra keyword arguments with `**kwargs`, and "mlab" isn't particularly informative about where the kwargs are actually going.

I'd also like to minimize changes unrelated to this specific enhancement. There is some inconsistency between pysurfer parameter names and mayavi ones, but aiming to fully streamline that would be a rabbit hole that is best to avoid.

Reply:
Agreed, I would just do:
```
**kwargs : additional keyword arguments
    These are passed to the underlying :func:`mayavi.mlab.surface` call.
```
To figure out the links you can dump the `intersphinx` inventory with something like:
```
$ python -msphinx.ext.intersphinx http://docs.enthought.com/mayavi/mayavi/objects.inv > mayavi.txt
```
Then browse for relevant entries in `mayavi.txt`. Unfortunately they might not have entries for each function you actually want to use.

Reply:
I have updated the documents, but unfortunately, those for the functions in `mlab.pipeline` are missing.